pkg/conf/conf.go

```diff
@@ -381,48 +382,7 @@ type Poller struct {
 }
 
 func (p *Poller) Union(defaults *Poller) {
-	if p.Datacenter == nil && defaults.Datacenter != nil {
-		p.Datacenter = defaults.Datacenter
-	}
-	if p.Addr == nil && defaults.Addr != nil {
-		p.Addr = defaults.Addr
-	}
-	if p.AuthStyle == nil && defaults.AuthStyle != nil {
-		p.AuthStyle = defaults.AuthStyle
-	}
-	if p.Username == nil && defaults.Username != nil {
-		p.Username = defaults.Username
-	}
-	if p.Password == "" && defaults.Password != "" {
-		p.Password = defaults.Password
-	}
-	if p.UseInsecureTls == nil && defaults.UseInsecureTls != nil {
-		p.UseInsecureTls = defaults.UseInsecureTls
-	}
-	if p.SslCert == nil && defaults.SslCert != nil {
-		p.SslCert = defaults.SslCert
-	}
-	if p.SslKey == nil && defaults.SslKey != nil {
-		p.SslKey = defaults.SslKey
-	}
-	if p.LogMaxBytes == nil && defaults.LogMaxBytes != nil {
-		p.LogMaxBytes = defaults.LogMaxBytes
-	}
-	if p.LogMaxFiles == nil && defaults.LogMaxFiles != nil {
-		p.LogMaxFiles = defaults.LogMaxFiles
-	}
-	if p.Exporters == nil && defaults.Exporters != nil {
-		p.Exporters = defaults.Exporters
-	}
-	if p.Collectors == nil && defaults.Collectors != nil {
-		p.Collectors = defaults.Collectors
-	}
-	if p.IsKfs == nil && defaults.IsKfs != nil {
-		p.IsKfs = defaults.IsKfs
-	}
-	if p.PollerSchedule == nil && defaults.PollerSchedule != nil {
-		p.PollerSchedule = defaults.PollerSchedule
-	}
+	_ = mergo.Merge(p, defaults)
```

Comment:
nice. Means no more reflection code needed as well?

Reply:
More accurate to say we moved the reflection from our code to theirs 😄
but that's still a win. We also still need to keep `Union2` until we remove node from config code.